File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end());
        
        int i = 0, j = people.size()-1;
        
        int minNumBoats = 0;
        
        while(i <= j){
            minNumBoats++;
            if(people[i]+people[j] <= limit){
                i++;
            }
            j--;
        }
        
        return minNumBoats;
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int n = people.length;
        int boats = 0;
        int left = 0, right = n - 1;
        while(left < right){
            if((people[left] + people[right]) > limit){
                right--;
            } else{
                left++; 
                right--;
            }
            boats++;
        }
        if(left == right) boats++;
        return boats;
    }
}